pkg/yang/entry.go

```diff
@@ -501,7 +501,10 @@ func semCheckMaxElements(v *Value) (uint64, error) {
 	}
 	val, err := strconv.ParseUint(v.Name, 10, 64)
 	if err != nil {
-		return val, fmt.Errorf(`%s: invalid max-elements value %q (expect "unbounded" or a non-negative integer): %v`, Source(v), v.Name, err)
+		return val, fmt.Errorf(`%s: invalid max-elements value %q (expect "unbounded" or a positive integer): %v`, Source(v), v.Name, err)
+	}
+	if val == 0 {
+		return val, fmt.Errorf(`%s: invalid max-elements value %q (expect "unbounded" or a positive integer)`, Source(v), v.Name)
```

Comment:
Should we make this error message different to the one @504 -- it seems like them being the same might cause us some extra debugging effort in the future.